metaflow/cli.py

```diff
@@ -765,6 +766,8 @@ def start(ctx,
                                                   ctx.obj.event_logger,
                                                   ctx.obj.monitor)
     ctx.obj.datastore = DATASTORES[datastore]
+    if datastore_root is None:
+        datastore_root = ctx.obj.datastore.get_datastore_root_from_config(echo)
```

Comment:
Discussed offline - I think each of the terminal commands like `step`, `dump` etc. probably handle if `datastore_root is None` so we should be able to safely remove this `if` check.

Reply:
Made the change.